server/controller/message/communityMessage.js

```diff
@@ -0,0 +1,57 @@
+import models from '../../models';
+import { characterLengthCheck, linkCheck } from '../../utils/messageCheck';
+
+const { User, CommunityMessage } = models;
+
+
+/**
+ * A class that handles user profile operations
+* */
+class CommunityMessageController {
+  /**
+     * post a message to the community
+     * @param {object} req - request object
+     * @param {object} res - response object
+     * @param{function} next - next function
+     * @returns {object} response object
+     *
+     */
+  static async postMessage(req, res, next) {
+    try {
+      const { userId } = req.user;
+      let isApproved = true;
+      const user = await User.findByPk(userId);
+      const { body } = req.body;
+      if (characterLengthCheck(body) || linkCheck(body)) {
+        isApproved = false;
+      }
```

Comment:
On second thought here - since we're persisting the `isApproved` field, we can set its `defaultValue` to `true` on the `CommunityMessage` model definition, and instead of having line 22, 25 to 27 we do the below instead.

```js
 const isApproved = !(characterLengthCheck(body) || linkCheck(body))
```
Then we get rid of lines 22, 25 to 27, and set the `defaultValue` of `isApproved` to `true` on the CommunityMessage model definition and migrations.

@chekwas88 please let me know if this makes sense

Reply:
Makes a lot of sense